tck/src/main/java/org/eclipse/microprofile/lra/tck/TckCancelOnTests.java

```diff
@@ -0,0 +1,198 @@
+                beforeCompensatedCount + 1, getCompensateCount());
+        assertEquals("After status code 303 is received, complete can't be invoked as not defined in annotation @LRA",
+                beforeCompletedCount, getCompletedCount());
+    }
+
+    /**
+     * See {@link LraCancelOnController#cancelOn301()}
+     */
+    @Test
+    public void cancelOn301() {
+        Response response = getSuiteTarget().path(LRA_CANCEL_ON_CONTROLLER_PATH)
+                .path(LraCancelOnController.CANCEL_ON_301).request().get();
+        assertEquals("The 301 status response is expected", Status.MOVED_PERMANENTLY.getStatusCode(), response.getStatus());
+        assertEquals("After status code 301 is received, compensate should be invoked as set by attribute cancelOn",
+                beforeCompensatedCount + 1, getCompensateCount());
+        assertEquals("After status code 301 is received, complete can't be invoked as not defined in annotation @LRA",
+                beforeCompletedCount, getCompletedCount());
+    }
+
+    /**
+     * See {@link LraCancelOnController#notCancelOnFamily5xx()}
+     */
+    @Test
+    public void notCancelOnFamily5xx() {
+        Response response = getSuiteTarget().path(LRA_CANCEL_ON_CONTROLLER_PATH)
+                .path(LraCancelOnController.NOT_CANCEL_ON_FAMILY_5XX).request().get();
+        assertEquals("The 500 status response is expected", Status.INTERNAL_SERVER_ERROR.getStatusCode(), response.getStatus());
+        assertEquals("After status code 500 is received, compensate can't be invoked as default behaviour was changed",
+                beforeCompensatedCount, getCompensateCount());
+        assertEquals("After status code 500 is received, complete has to be called as default behaviour was changed",
+                beforeCompletedCount + 1, getCompletedCount());
+    }
+
+    /**
+     * See {@link LraCancelOnController#cancelFromRemoteCall()}
+     */
+    @Test
+    public void cancelFromRemoteCall() {
+        Response response = getSuiteTarget().path(LRA_CANCEL_ON_CONTROLLER_PATH)
+                .path(LraCancelOnController.CANCEL_FROM_REMOTE_CALL).request().get();
+        assertEquals("The 200 status response is expected", Status.OK.getStatusCode(), response.getStatus());
+        assertEquals("Status was 200 but compensate should be called twice as LRA should be cancelled for remotely called participant as well",
+                beforeCompensatedCount + 2, getCompensateCount());
+        assertEquals("Even the 200 status was received the remotly called participant should cause the LRA being cancelled",
+                beforeCompletedCount, getCompletedCount());
+    }
+
+
+
+    private int getCompletedCount() {
+        Response response = getSuiteTarget().path(LRA_CANCEL_ON_CONTROLLER_PATH)
+                .path(LraCancelOnController.COMPLETED_COUNT_PATH).request().get();
+        return Integer.valueOf(response.readEntity(String.class));
```

Comment:
you can also use directly `response.readEntity(Integer.class)`

Reply:
thanks!